app/federation/server/PeerDNS.js

```diff
@@ -95,20 +95,23 @@ export class PeerDNS {
 
 		const [srvEntry] = srvEntries;
 
-		// Get the public key from the TXT record
-		const txtRecords = dnsResolveTXT(`rocketchat-public-key.${ domain }`);
+		// Get the protocol from the TXT record, if exists
+		let protocol = 'https';
 
-		// Get the first TXT record, this subdomain should have only a single record
-		const txtRecord = txtRecords[0];
+		try {
+			const protocolTxtRecords = dnsResolveTXT(`rocketchat-protocol.${ domain }`);
 
-		// If there is no record, skip
-		if (!txtRecord) {
-			throw new Meteor.Error('ENOTFOUND', 'Could not find public key entry on TXT records');
+			protocol = protocolTxtRecords[0][0].toLowerCase() === 'http' ? 'http' : 'https';
```

Comment:
is [0][0] always guaranteed to be here?  Should we check here or are we ok with this pattern of just "handling" / ignoring the exception that is thrown?

Reply:
from node docs it seems to br guaranteed